jwql/instrument_monitors/common_monitors/bias_monitor.py

```diff
@@ -0,0 +1,554 @@
+            2D array on which to calculate statistics
+
+        Returns
+        -------
+        collapsed_rows : numpy.ndarray
+            1D array of the collapsed row values
+        
+        collapsed_columns : numpy.ndarray
+            1D array of the collapsed column values
+        """
+
+        collapsed_rows = np.nanmedian(image, axis=1)
+        collapsed_columns = np.nanmedian(image, axis=0)
+
+        return collapsed_rows, collapsed_columns
+
+    def extract_zeroth_group(self, filename):
+        """Extracts the 0th group of a fits image and outputs it into
+        a new fits file.
+        
+        Parameters
+        ----------
+        filename : str
+            The fits file from which the 0th group will be extracted.
+
+        Returns
+        -------
+        output_filename : str
+            The full path to the output file
+        """
+
+        output_filename = os.path.join(self.data_dir,
+            os.path.basename(filename).replace('.fits', '_0thgroup.fits'))
+        
+        # Write a new fits file containing the primary and science
+        # headers from the input file, as well as the 0th group
+        # data of the first integration
+        if not os.path.isfile(output_filename):
+            hdu = fits.open(filename)
+            new_hdu = fits.HDUList([hdu['PRIMARY'], hdu['SCI']])
+            new_hdu['SCI'].data = hdu['SCI'].data[0:1, 0:1, :, :]
+            new_hdu.writeto(output_filename)
+            hdu.close()
+            new_hdu.close()
+            logging.info('\t{} created'.format(output_filename))
+        else:
+            logging.info('\t{} already exists'.format(output_filename))
+            pass
+
+        return output_filename
+
+    def file_exists_in_database(self, filename):
+        """Checks if an entry for filename exists in the bias stats 
+        database.
+
+        Parameters
+        ----------
+        filename : str
+            The full path to the uncal filename
+
+        Returns
+        -------
+        file_exists : bool
+            ``True`` if filename exists in the bias stats database
+        """
+
+        query = session.query(self.stats_table)
+        results = query.filter(self.stats_table.uncal_filename == filename).all()
+
+        if len(results) != 0:
+            file_exists = True
+        else:
+            file_exists = False
+
+        return file_exists
+
+    def get_amp_medians(self, image, amps):
+        """Calculates the median in the input image for each amplifier
+        and for odd and even columns separately.
+
+        Parameters
+        ----------
+        image : numpy.ndarray
+            2D array on which to calculate statistics
+
+        amps : dict
+            Dictionary containing amp boundary coordinates (output from
+            ``amplifier_info`` function)
+            ``amps[key] = [(xmin, xmax, xstep), (ymin, ymax, ystep)]``
+
+        Returns
+        -------
+        amp_meds : dict
```

Comment:
I suggest to change this variable name to `amp_medians`.  It's better to be explicit about these things when possible.

Reply:
fixed